src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/H2Engine.java

```diff
@@ -147,7 +153,16 @@ protected int entityToPreparedStatement(final DbEntity entity, final PreparedSta
 
                 switch (column.getDbColumnType()) {
                     case BLOB:
-                        ps.setBytes(i, objectToArray(val));
+                        if (val == null) {
+                            ps.setNull(i, Types.BLOB);
+                            break;
+                        }
+
+                        if (val instanceof Serializable) {
```

Comment:
I would use an `else if` here and remove the break from the block above...

Reply:
I don't think we even need to have the special case for null here....
Check H2 driver internals, it seems to handle _null_ just fine

Reply:
The problem here is not the driver, but the `entry` map. This map may not have the key, so it is necessary to check for nulls.